werkzeug/_internal.py

```diff
@@ -282,8 +284,8 @@ def _cookie_parse_impl(b):
         if not match:
             break
 
-        key = match.group('key').strip()
-        value = match.group('val')
+        key = (match.group('key') or '').strip()
+        value = match.group('val').strip()
```

Comment:
It appears that stripping shouldn't be needed at all since the regex ought to take care of this.